src/sentry/api/endpoints/issues/related_issues.py

```diff
@@ -33,14 +41,19 @@ def get(self, request: Request, group: Group) -> Response:
         :pparam Request request: the request object
         :pparam Group group: the group object
         """
-        # The type of related issues to retrieve. Can be either `same_root_cause` or `trace_connected`.
-        related_type = request.query_params["type"]
-        extra_args = {
-            "event_id": request.query_params.get("event_id"),
-            "project_id": request.query_params.get("project_id"),
-        }
+        serializer = RequestSerializer(data=request.query_params)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        _data = serializer.validated_data
+        related_type = _data["type"]
         try:
-            data, meta = RELATED_ISSUES_ALGORITHMS[related_type](group, extra_args)
+            data, meta = (
+                same_root_cause_analysis(group)
+                if related_type == "same_root_cause"
+                else trace_connected_analysis(
+                    group, event_id=_data.get("event_id"), project_id=_data.get("project_id")
+                )
+            )
             return Response({"type": related_type, "data": data, "meta": meta})
         except AssertionError:
```

Comment:
unrelated to this PR but `except AssertionError` is something that should really only be in tests -- assertion errors shouldn't really be used at runtime since they're meant to be "programmer assertions" and not business logic